lib/adapter.js

```diff
@@ -2419,8 +2425,9 @@ function Adapter(options) {
                 callback = options;
                 options = undefined;
             }
-            if (!adapterObjects) { // if objects is no longer existing, we do not need to unsubscribe
-                return tools.maybeCallbackWithError(callback, 'Objects database not connected');
+            if (!adapterObjects) {
+                this.log.info('getObjectView not processed because Objects database not connected');
```

Comment:
Should we also log this, even if cb is present? Because if cb is present, the error will be passed to it and the adapter dev has a chance to react to it.

Reply:
thats why I decided to always log here. In fact the case that adapterStates/OBjects are NO longer defined means that some code is executed BEFORE ready or AFTER unload (or unload too too long). And this is a clear issue in any means (irrelevant if promise or callback)